distribution/docker/alpine/Dockerfile

```diff
@@ -1,20 +1,53 @@
-FROM docker.io/alpine:3.14 AS builder
+ARG ALPINE_IMAGE=docker.io/alpine:3.15
 
-WORKDIR /vector
+#-------------------------------------------------------------------------------
+FROM ${ALPINE_IMAGE} AS staging
 
-COPY vector-*-unknown-linux-musl*.tar.gz ./
-RUN tar -xvf vector-0*-"$(cat /etc/apk/arch)"-unknown-linux-musl*.tar.gz --strip-components=2
+# Set a well-known working directory for staging
+WORKDIR /staging
 
-RUN mkdir -p /var/lib/vector
+# Prepare a staging root filesystem
```

Comment:
Clever way to easily only copy what is needed into the image.

Reply:
All credit here goes to @hhromic!